File: src/main/java/com/example/configcenter/entity/Ticket.java
package com.example.configcenter.entity;

import com.example.configcenter.enums.TicketActionEnum;
import com.example.configcenter.enums.TicketPhaseEnum;
import lombok.Data;

import java.util.List;

/**
 * 工单实体类
 */
@Data
public class Ticket {

    /**
     * Ticket ID
     */
    private Long id;

    /**
     * 关联数据ID
     */
    private Long dataId;

    /**
     * 工单标题
     */
    private String title;

    /**
     * 工单阶段
     */
    private TicketPhaseEnum phase;

    /**
     * 申请人
     */
    private String applicator;

    /**
     * 操作人
     */
    private String operator;

    /**
     * 旧数据(JSON格式)
     */
    private String oldData;

    /**
     * 新数据(JSON格式)
     */
    private String newData;

    /**
     * 创建时间
     */
    private long createTime;

    /**
     * 更新时间
     */
    private long updateTime;

    /**
     * 行为列表
     */
    private List<TicketActionEnum> action;

    public List<TicketActionEnum> getAction() {
        switch (phase) {
            case Reviewing:
                return List.of(TicketActionEnum.Approve, TicketActionEnum.Reject, TicketActionEnum.Cancel);
            case GrayPublish:
                return List.of(TicketActionEnum.Publish, TicketActionEnum.Cancel);
            default:
                return List.of();
        }
    }
}

File: frontend/src/types/ticket.ts
// 工单状态枚举
export enum TicketPhase {
  Submit = 'Submit',
  Reviewing	 = 'Reviewing',
  Rejected = 'Rejected',
  Success = 'Success',
  GrayPublish = 'GrayPublish',
  Cancelled = 'Cancelled'
}

// 工单实体
export interface Ticket {
  id: number
  dataId: number
  title: string
  phase: TicketPhase
  applicator: string
  operator?: string
  createTime: long
  updateTime: long
  createBy?: string
  updateBy?: string
  oldData?: string
  newData?: string
}

// 工单查询请求
export interface TicketQueryRequest {
  pageNum: number
  pageSize: number
  title?: string
  phase?: string
  applicator?: string
}

// 工单创建请求
export interface TicketCreateRequest {
  dataId: number
  title: string
  newData: string
}

// 工单更新请求
export interface TicketUpdateRequest {
  id: number
  title?: string
  phase?: TicketPhase
  operator?: string
  newData?: string
} 